pkg/cluster/storage.go

```diff
@@ -151,7 +150,13 @@ func bootstrapKeyData(ctx context.Context, storageClient client.Client) (*client
 // bootstrap key as a lock. This function will not return successfully until either the
 // bootstrap key has been locked, or data is read into the struct.
 func (c *Cluster) storageBootstrap(ctx context.Context) error {
-	if err := c.startStorage(ctx); err != nil {
+	bootstrapCtx, cancel := context.WithCancel(ctx)
+	defer func() {
+		cancel()
+	}()
+
+	logrus.Info("Starting temporary kine to bootstrap the storage")
```

Comment:
nit: consistent with message at https://github.com/k3s-io/k3s/blob/1c8be1d0111e45f188d35d6d3c35098daa0544a3/pkg/cluster/bootstrap.go#L485
```suggestion
	logrus.Info("Starting temporary kine to reconcile with datastore")
```

Reply:
for this part, there is another one that made me think

```go
        bootstrapCtx, cancel := context.WithCancel(ctx)
	defer func() {
		cancel()
	}()

	if c.config.KineTLS {
		logrus.Info("Starting temporary kine to reconcile with datastore")
		if err := c.startStorage(bootstrapCtx, true); err != nil {
			return err
		}
	} else {
		if err := c.startStorage(ctx, true); err != nil {
			return err
		}
	}
```

I decided to add a if here, because I believe we do not want 2 kine instances being made when we don't want it, what do you think @brandond?

Reply:
I would probably move the context setup inside that function too, but yeah looks good otherwise.